climakitae/explore/typical_meteorological_year.py

```diff
@@ -0,0 +1,804 @@
+        """Get coordinates and other station metadata from station
+
+        Parameters
+        ----------
+        station_name: str
+           Name of HadISD station
+        """
+        # read in station file of CA HadISD stations
+        stn_file = pkg_resources.resource_filename(
+            "climakitae", "data/hadisd_stations.csv"
+        )
+        stn_file = pd.read_csv(stn_file, index_col=[0])
+        # grab airport
+        try:
+            self.stn_name = station_name
+            self.stn_code = stn_file.loc[stn_file["station"] == self.stn_name][
+                "station id"
+            ].item()
+            one_station = stn_file.loc[stn_file["station"] == self.stn_name]
+        except ValueError as e:
+            raise ValueError(
+                "Could  not find station in hadisd_stations.csv. Please provide valid station name."
+            ) from e
+        self.stn_lat = one_station.LAT_Y.item()
+        self.stn_lon = one_station.LON_X.item()
+        self.stn_state = one_station.state.item()
+        self._set_lat_lon()
+        return
+
+    def _set_loc_from_lat_lon(self, latitude: float, longitude: float):
+        """Set class attributes based on latitude and longitude variables.
+
+        Parameters:
+        -----------
+        latitude: float
+           Location latitude in degrees
+        longitude: float
+           Location longitude in degrees
+        """
+        self.stn_lat = latitude
+        self.stn_lon = longitude
+        self._set_lat_lon()
+        # Set station variables to string "None"
+        # to be written to file with final data
+        self.stn_name = "None"
+        self.stn_code = "None"
+        self.stn_state = "None"
+        return
+
+    def _set_lat_lon(self):
+        """Set the lat/lon ranges for selecting data around the grid point."""
+        self.latitude = (self.stn_lat - 0.05, self.stn_lat + 0.05)
+        self.longitude = (self.stn_lon - 0.06, self.stn_lon + 0.06)
+        return
+
+    def _vprint(self, msg: str):
+        """Checks verbosity and prints as allowed."""
+        if self.verbose:
+            print(msg)
+        return
+
+    def _load_single_variable(self, varname: str, units: str) -> xr.DataArray:
+        """Fetch catalog data for one variable.
+
+        Parameters
+        ----------
+        varname: str
+           Name of desired catalog variable
+        units: str
+           Desired units
+
+        Returns
+        -------
+        xr.DataArray
+        """
+        # Extra year in UTC time to get full period in local time.
+        if self.end_year == 2100:
+            print(
+                "End year is 2100. The final day in timeseries may be incomplete after data is converted to local time."
+            )
+            new_end_year = self.end_year
+        else:
+            new_end_year = self.end_year + 1
+
+        data = get_data(
+            variable=varname,
+            resolution="9 km",
```

Comment:
We should realistically bump down to 3 km (an update from the OG notebook). We'll have to re-test timing to see how that changes and whether the trade off is worth it.